### ISSUE TITLE
Warning: foreach() argument must be of type array|object, string given

### ISSUE THREAD

**Issue by laryn** [backdrop-contrib/fivestar#19]
> Warning: foreach() argument must be of type array|object, string given in fivestar_get_tags() (line 843 of /var/www/html/backdrop/modules/contrib/fivestar/fivestar.module).

Received when adding a new Fivestar field to a content type.

**Comment by albanycomputers** [comment]
Line 840 of fivestar.module dev version... is causing an error  

```explode(): Argument #2 ($string) must be of type string, array given in explode() (line 840 of fivestar.module).```

But, looking at the 2017 version 2.2 there's no difference... So, to say "I'm lost" would be an understatement...

But, explode from what the manual says, splits text based on a character... in this case a comma...

but in the settings file...  fivestar.settings.json  saved in the module, the "fivestar_tags": ["vote"] is an array... 

Experimenting, I added to the tag... and could not replicate the error until I examined the settings file saved in core/configxxx/active

When you add a new value to the tag in /admin/config/content/fivestar i.e.  "vote,rate" the value is saved as text and not an array...
I removed the "rate" tag... again, just experimenting... and the settings fivestar.settings.json is now "fivestar_tags": "vote"

no longer an array.

So, is the error... the fact the initial settings fivestar.settings.json is "fivestar_tags": ["vote"] when it should be "fivestar_tags": "vote" ?

The variable... $tags_txt in
```$tags_txt = config_get('fivestar.settings', 'fivestar_tags');```
suggests it's expecting a text entry and not an array.
